packages/dds/tree/src/test/shared-tree/editing.spec.ts

```diff
@@ -3146,6 +3146,119 @@ describe("Editing", () => {
 			});
 		});
 
+		describe("Inverse preconditions", () => {
+			it("inverse constraint not violated by interim change", () => {
+				const tree = makeTreeFromJson({ foo: "A" });
+				const stack = createTestUndoRedoStacks(tree.events);
+
+				// Make transaction on a branch that does the following:
+				// 1. Changes value of "foo" to "B".
+				// 2. Adds inverse constraint on existence of node "B" on field "foo".
+				tree.transaction.start();
+				tree.editor
+					.valueField({ parent: rootNode, field: brand("foo") })
+					.set(singleJsonCursor("B"));
+				tree.editor.addNodeExistsConstraintOnRevert({
+					parent: rootNode,
+					parentField: brand("foo"),
+					parentIndex: 0,
+				});
+				tree.transaction.commit();
+				expectJsonTree(tree, [{ foo: "B" }]);
+
+				const changedFooAtoB = stack.undoStack[0] ?? fail("Missing undo");
+
+				// This change should not violate the constraint in the inverse because it is changing
+				// a different node on filed "bar".
+				tree.editor
+					.optionalField({ parent: rootNode, field: brand("bar") })
+					.set(singleJsonCursor("C"), true);
+
+				// This revert should apply since its constraint has not been violated
+				changedFooAtoB.revert();
+				expectJsonTree(tree, [{ foo: "A", bar: "C" }]);
+
+				stack.unsubscribe();
+			});
+
+			it("inverse constraint violated by interim change", () => {
```

Comment:
Nit: just being more specific about this test and the next one
```suggestion
			it("inverse constraint violated by a change between the original and the revert", () => {
```